applications/IgaApplication/tests/test_nurbs_volume_element.py

```diff
@@ -0,0 +1,202 @@
+
+class TestNurbsVolumeElement(KratosUnittest.TestCase):
+
+    '''
+    Test description:
+    Provides necessary methods to run test for Iga volume elements.
+    A simple cantilever beam is computed and checked against the analytical solution.
+    '''
+    @classmethod
+    def create_geometry(cls, model_part, order_u, order_v_w, length):
+        nodes = KM.NodesVector()
+        # Create control points
+        node_id = 1
+        width = 1.0
+        delta_u = length / order_u
+        delta_v = width / order_v_w
+        delta_w = width / order_v_w
+        z_list = [x for x in decimal_range(0.0, width + 0.5*delta_w, delta_w)]
+        y_list = [x for x in decimal_range(0.0, width + 0.5*delta_v, delta_v)]
+        x_list = [x for x in decimal_range(0.0, length + 0.5*delta_u, delta_u)]
+        for k in z_list: #z-direction
+            for j in y_list: #y-direction
+                for i in x_list: #x-direction
+                    node = model_part.CreateNewNode(node_id, i, j, k)
+                    nodes.append(node)
+                    node_id = node_id + 1
+
+        knots_u = KM.Vector(2*order_u)
+        for i in range(2*order_u):
+            if( i < order_u):
+                knots_u[i] = 0.0
+            else:
+                knots_u[i] = 1.0
+
+        knots_v = KM.Vector(2*order_v_w)
+        knots_w = KM.Vector(2*order_v_w)
+        for i in range(2*order_v_w):
+            if( i < order_v_w):
+                knots_v[i] = 0.0
+                knots_w[i] = 0.0
+            else:
+                knots_v[i] = 1.0
+                knots_w[i] = 1.0
+
+        volume = KM.NurbsVolumeGeometry(
+            nodes, order_u, order_v_w, order_v_w, knots_u, knots_v, knots_w)
+
+        return volume
+
+    @classmethod
+    def solve_cantilever(cls, order_u, order_v_w, length):
+        model = KM.Model()
+        model_part = model.CreateModelPart('Model')
+
+        model_part.AddNodalSolutionStepVariable(KM.DISPLACEMENT)
+        model_part.AddNodalSolutionStepVariable(KM.REACTION)
+        model_part.AddNodalSolutionStepVariable(StructuralMechanicsApplication.POINT_LOAD)
+        model_part.ProcessInfo.SetValue(KM.DOMAIN_SIZE,3)
+
+        # Create property for volume elements
+        cl = StructuralMechanicsApplication.LinearElastic3DLaw()
+        volume_properties = model_part.GetProperties()[0]
+
+        volume_properties.SetValue(KM.YOUNG_MODULUS, 210000000)
+        volume_properties.SetValue(KM.POISSON_RATIO, 0.3)
+        volume_properties.SetValue(KM.DENSITY, 78.5)
+        volume_properties.SetValue(KM.CONSTITUTIVE_LAW, cl)
+
+        # Create a nurbs volume
+        volume = TestNurbsVolumeElement.create_geometry(model_part, order_u, order_v_w, length)
+
+        # Create quadrature_point_geometries
+        quadrature_point_geometries = KM.GeometriesVector()
+
+        volume.CreateQuadraturePointGeometries(quadrature_point_geometries, 2)
+
+        for i in range(len(quadrature_point_geometries)):
```

Comment:
For some reason the enumerate or a standard for loop on the quadrature point geometries does not work. There is an error with the ownership between python and C++. To make it work I would have to change the 'self.begin()' to 'self.ptr_begin()' in the following line....

https://github.com/KratosMultiphysics/Kratos/blob/4216e1312c737825b58acd60e40abd90227762c0/kratos/python/containers_interface.h#L94

Reply:
Ok. I've never used it myself but I found it interesting. Nevertheless that's absolutely minor.